content/04.model-complexity.regularization.md

```diff
@@ -0,0 +1,55 @@
+Regularization makes models less sensitive to variance in sample data by adding a penalty (lambda) to the sum of squared residuals (or sum of maximum likelihoods). 
+While the value of lambda is determined by crossvalidation, the minimization of the sum of squared residuals (or error) leads to a model that is less sensitive to changes in sample data.
+
+To learn from 'n' number of features, machine learning models ideally need at least 'n+1' samples. 
+For eg. to predict any disease outcome from 20,000 genes we need at least 20,001 samples. 
+However, in reality even the best sample set probably contains 500 samples, and in case of rare diseases that may be closer to 50 than 500. 
+Regularization can not only minimize overfitting but can additionally help in predicting outcomes using fewer number of samples. 
+Regularization can be of three main types, each with their particular strengths and weaknesses. 
+(1) Ridge regression aims to minimize the magnitude of the features, but in models that try to select the most important features for accurate prediction of sample labels, ridge regression shrinks all features equally, but can never shrink features all the way to 0. 
+Thus in presence of many correlated parameters (eg. gene expression networks), ridge regression may not be ideal in reducing the feature space. 
+(2) LASSO regression on the other hand works well for selecting few important features since its effect can shrink some features more than the others. 
+Thus it helps in selecting most important features while irrelevant features are lost and shrunk to 0. This selection attribute of LASSO (in a sample set of size "n", LASSO can only select "n" features for the model) may be an advantage in reducing model complexity, but a disadvantage in cases where identification of all possible collinear features is important (eg. all biomarkers correlating to a particular disease phenotype) [@doi:10.1038/nmeth.4014]. 
+(3) Elastic net regression is a combination of LASSO and ridge regression[@doi:10.1111/j.1467-9868.2005.00503.x]. 
+Both of the methodologies when applied together helps to select most useful features, specially where there are a lot of correlated features. 
+In this setup, LASSO leads to selection of one of the correlated features and reduces the others to 0 (grouping of features). 
+The ridge regression aspect of elastic-net then minimizes the magnitude of the features that were selected without shrinking any of them to 0 (minimization). 
+
+Any supervised learning implementation in rare disease would require robustness towards feature selection from a small number of samples. 
+This robustness is mostly acquired through the combination of various regression strategies. 
+Since machine learning applications in rare disease is infrequent, combination strategies used for rare variant discovery and immune cell signature discovery can serve as good case studies to examine. 
```

Comment:
```suggestion
Since machine learning applications in rare disease are infrequent, combination strategies used for rare variant discovery and immune cell signature discovery can serve as good case studies to examine. 
```